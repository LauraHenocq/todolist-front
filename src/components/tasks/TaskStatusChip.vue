<script setup lang="ts">
import { TaskStatus } from '@/models/task.entity';
import { computed } from 'vue';

type Props = {
  status: TaskStatus
}

const props = defineProps<Props>()

const getStatus = (): string => {
  switch (props.status) {
    case TaskStatus.TODO:
      return 'A faire'
    case TaskStatus.IN_PROGRESS:
      return 'En cours'
    case TaskStatus.DONE:
      return 'TerminÃ©e'
    default:
      return 'A faire'
  } 
}

const getStatusColor = computed(() => {
  switch (props.status) {
  case TaskStatus.TODO:
      return 'red'
    case TaskStatus.IN_PROGRESS:
      return 'secondary'
    case TaskStatus.DONE:
      return 'green'
    default:
      return 'grey'
  }
});
</script>

<template>
  <v-chip variant="flat" :color="getStatusColor">{{ getStatus(props.status) }}</v-chip>
</template>