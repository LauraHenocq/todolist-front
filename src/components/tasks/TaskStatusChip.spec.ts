import { mount, VueWrapper } from '@vue/test-utils';
import { nextTick } from 'vue';
import { beforeEach, expect, it, describe } from 'vitest'
import { createVuetify } from 'vuetify';
import TaskStatusChip from './TaskStatusChip.vue';
import { TaskStatus } from '@/models/task.entity';

const vuetify = createVuetify();

describe('TaskStatusChip', () => {
  let wrapper: VueWrapper
  const status = TaskStatus.TODO;

  const defaultProps = {
    status
  }

  beforeEach(() => {
    wrapper = mount(TaskStatusChip, {
      props: defaultProps,
      global: {
        plugins: [vuetify],
      },
    });
  });

  describe('render', () => {
    it('should render', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });
  });
});