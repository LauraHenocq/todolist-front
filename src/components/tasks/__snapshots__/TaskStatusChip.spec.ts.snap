// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TaskStatusChip > render > should render 1`] = `
"<span class="v-chip v-theme--light bg-red v-chip--density-default v-chip--size-default v-chip--variant-flat" draggable="false"><!----><span class="v-chip__underlay"></span>
<!---->
<!---->
<div class="v-chip__content" data-no-activator="">A faire</div>
<!---->
<!----></span>"
`;
